src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpUseCollectionExpressionForArrayCodeFixProvider.cs

```diff
@@ -45,84 +44,111 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
         return Task.CompletedTask;
     }
 
-    protected override async Task FixAllAsync(
+    protected sealed override async Task FixAllAsync(
         Document document,
         ImmutableArray<Diagnostic> diagnostics,
         SyntaxEditor editor,
         CodeActionOptionsProvider fallbackOptions,
         CancellationToken cancellationToken)
     {
-        var sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
fixed this up to follow the pattern we use for all the rest of the rewriters.  important, this handles successive rewrites with arrays within arrays.